gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -957,6 +959,52 @@ gnc_plugin_page_register_ui_update (gpointer various, GncPluginPageRegister *pag
                                          "UnvoidTransactionAction");
     gtk_action_set_sensitive (GTK_ACTION(action), voided);
 
+    /* Set available actions based on read only */
+    if (xaccTransGetReadOnly (trans) || xaccTransIsReadonlyByPostedDate (trans))
+        read_only = TRUE;
+
+    action = gnc_plugin_page_get_action (GNC_PLUGIN_PAGE(page),
+                                         "CutTransactionAction");
+    gtk_action_set_sensitive (GTK_ACTION(action), !read_only);
+
+    action = gnc_plugin_page_get_action (GNC_PLUGIN_PAGE(page),
+                                         "PasteTransactionAction");
+    gtk_action_set_sensitive (GTK_ACTION(action), !read_only);
+
+    action = gnc_plugin_page_get_action (GNC_PLUGIN_PAGE(page),
+                                         "DeleteTransactionAction");
+    gtk_action_set_sensitive (GTK_ACTION(action), !read_only);
+
+    action = gnc_plugin_page_get_action (GNC_PLUGIN_PAGE(page),
+                                         "DuplicateTransactionAction");
+    gtk_action_set_sensitive (GTK_ACTION(action), !read_only);
+
+    action = gnc_plugin_page_get_action (GNC_PLUGIN_PAGE(page),
+                                         "RemoveTransactionSplitsAction");
+    gtk_action_set_sensitive (GTK_ACTION(action), !read_only);
+
+    /* Set available split actions based on split account anchor */
+    if (cursor_class == CURSOR_CLASS_SPLIT)
+    {
+        Split *split = gnc_split_register_get_current_split (reg);
+        gboolean disable = FALSE;
+
+        if (split == gnc_split_register_get_current_trans_split (reg, NULL))
+            disable = TRUE;
```

Comment:
How will this logic behave if the current transaction has more than one split in the current account register ?

Reply:
If the transaction has more than one split in the current account register you are able to delete one of them but not the other based on the order loaded in the register from the split list used to create the register.

Reply:
That's what I suspected and what I think is suboptimal. I think if there is more than one split in the current register the user should either be prevented from deleting any of those splits or should be allowed to delete any except of the last one (which may not be the anchoring split).
Remember our goal is to avoid the transaction from unexpectedly disappearing from the current register if the user deletes/cuts a split.
Oh but wait, if a transaction has more than one split in a register it will appear twice in that register, right ? So each of these two appearances has its own anchor split and removing the anchor split would suddenly drop that instance of the transaction from the register view... Darn complicated.

Regardless I did test this and if you don't know why the buttons are greyed out it's indeed pretty confusing you can delete one and not the other split.

I was thinking of dynamically updating tooltips for the menu items and toolbar buttons. That way a disabled button could have a tooltip that explains why it's disabled (like "Can't delete the anchoring split of a transaction). But currently tooltips for greyed out menu items are not shown in the status bar. Do you think we can have this or is this a Gtk limitation ?

Reply:
A transaction with two splits in the current account appears twice in Basic View, but one obviously can't edit the splits. In split view the transaction displays only once with all of its splits.

Reply:
I tested the Auto-split ledger, not the Transaction journal mode. Transaction journal mode behaves as you describe indeed.

Unfortunately that means the transaction journal mode does expose the poor behavior I tried to explain in my previous comment. Auto-split ledger is confusing in a different way as it may seem arbitrary which split you can delete in which instance of the same transaction.

I fear this to be a recipe for an pile of user support questions on what we developers perceive as perfectly logical from a technical point of view but will be non-obvious to the end-user... At least assuming many users actually use the cut/copy/delete split functionality.

Reply:
We could keep using the message box and block method instead of disabling the actions. That at least explains what's going on.

Reply:
Maybe then a combination of the two, grey out the options for when the transaction is read only / void and then add the warning to the cut / paste operation and also remove it from the General Journal.